src/transport/GroupPeerMessageCounter.cpp

```diff
@@ -277,9 +278,8 @@ CHIP_ERROR GroupOutgoingCounters::Init(chip::PersistentStorageDelegate * storage
     err = mStorage->SyncGetKeyValue(DefaultStorageKeyAllocator::GroupControlCounter().KeyName(), &temp, size);
     if (err == CHIP_ERROR_PERSISTED_STORAGE_VALUE_NOT_FOUND)
     {
-        // might be the first time we retrieve the value
-        // TODO handle this case
-        mGroupControlCounter = 0; // TODO should be random
+        // First time retrieving the counter
+        mGroupControlCounter = chip::Crypto::GetRandU32() & kMessageCounterRandomInitMask;
```

Comment:
This needs a +1.  The other code initializes to the value "one less than the next value we will return", while this code is initializing the actual next value it will return.